### DISCUSSION TITLE
Recognition of  specific phrase on mobile

### DISCUSSION THREAD

**Discussion by UEUEE** [openai/whisper#883] [Q&A] [answered]
Dear All, 

we are developing a language platform and one of the main features is training.  The user must pronounce a specific phrase (maybe several times) and the app should recognize it and mark recognized words in the sentence. 
Currently, we are using Pocketsphinx and it has such a function addGrammarSearch(grammar)
Vosk, another ASR also has similar 
recognizer = vosk.Model("path/to/model")
recognizer.SetKeyphrase("wakeup", "wake up")

The quality of recognition is lower in that case but it works really fast. 

Does Whisper have smth like that?
Is there any other way to make fast (without any delay) recognition with whisper?  

thanks!

**Comment by glangford** [comment, marked as answer]
See these existing discussions.
* https://github.com/openai/whisper/discussions/843
* https://github.com/openai/whisper/discussions/212
* https://github.com/openai/whisper/discussions/407
* https://github.com/openai/whisper/discussions/130
